### PR TITLE
object: add service account for RGW pod

### PR DESCRIPTION
**Pull request by thotz**
**Description of your changes:**
For supporting features like service account authentication for vault
KMS , a service account account need to attach with pod.

Signed-off-by: Jiffin Tony Thottan <thottanjiffin@gmail.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/apis/ceph.rook.io/v1/scc.go [hunk 1]
```diff
@@ -71,6 +71,7 @@ func NewSecurityContextConstraints(name, namespace string) *secv1.SecurityContex
 			fmt.Sprintf("system:serviceaccount:%s:default", namespace),
 			fmt.Sprintf("system:serviceaccount:%s:rook-ceph-mgr", namespace),
 			fmt.Sprintf("system:serviceaccount:%s:rook-ceph-osd", namespace),
+			fmt.Sprintf("system:serviceaccount:%s:rook-ceph-rgw", namespace),
```
Comment (travisn) [inline pkg/apis/ceph.rook.io/v1/scc.go:74]: The scc also needs this in operator-openshift.yaml

File: deploy/examples/common.yaml [hunk 2]
```diff
@@ -1236,6 +1248,20 @@ subjects:
...
 ---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: rook-ceph-rgw-psp
+  namespace: rook-ceph # namespace:cluster
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: psp:rook
```
Comment (travisn) [inline deploy/examples/common.yaml:1259]: psp is deprecated, can you remove this role binding? We need to look into removing other psps.

File: deploy/examples/common.yaml [hunk 3]
```diff
@@ -548,6 +548,18 @@ rules:
 ---
 kind: ClusterRole
```
Comment (travisn) [inline deploy/examples/common.yaml:549]: I would expect the rgw to use a `Role` instead of `ClusterRole`

File: deploy/examples/common.yaml [hunk 4]
```diff
@@ -548,6 +548,18 @@ rules:
...
 apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: rook-ceph-rgw
+rules:
+  - apiGroups: [""]
+    resources: ["configmaps"]
+    verbs: ["get", "list", "create", "patch", "update", "delete"]
```
Comment (travisn) [inline deploy/examples/common.yaml:556]: RGW is only the ceph daemon with no rook operator in the image, right? Is rook being added to the rgw image in the future or why are these privs needed for configmaps and secrets? The rgw daemon won't be accessing them directly, right?

**Review by travisn — CHANGES_REQUESTED** [review]

File: deploy/examples/common.yaml (hunk 3 above)
Comment (thotz) [reply deploy/examples/common.yaml:549]: Yaa that's my mistake.

File: deploy/examples/common.yaml (hunk 4 above)
Comment (thotz) [reply deploy/examples/common.yaml:556]: @travisn: currently Rook operator `PodSpec` for RGW which have configmap/secrets to need attach or create hence I have created this Rules. If it is not required then I will remove it. I was not 100% sure this role is needed. So most likely I need to create SA without any roles/rolebinding. Am I right??

Comment (travisn) [reply deploy/examples/common.yaml:556]: @thotz The pod actually doesn't need RBAC for the confimaps/secrets that are being mounted to the pod, it only needs the rbac if it is going to access the K8s API from a running container. Just creating the service with no roles/bindings sounds good.

**Comment by travisn** [comment]
LGTM, the rgw multisite test just seems to be failing, i've tried rerunning it a couple times...

**Comment by thotz** [comment]
> LGTM, the rgw multisite test just seems to be failing, i've tried rerunning it a couple times...

Apparently, I missed adding rgw SA into `common-secondary.yaml`

**Review by travisn — APPROVED** [review]

**Pull request merged**